python/pyarrow/tests/parquet/test_dataset.py

```diff
@@ -1239,6 +1241,71 @@ def test_dataset_partitioning(tempdir):
     assert result.column_names == ["a", "year", "month", "day"]
 
 
+@pytest.mark.parametrize(
```

Comment:
I think we should put this in the parent `test_dataset.py` (pyarrow/tests/test_dataset.py) since this isn't specific to ParquetDataset. Sorry I missed this earlier.